ATC/src/Plane.h

```diff
@@ -10,51 +10,59 @@
 
 #include <stdlib.h>
 #include <sys/neutrino.h>
+#include "commandCodes.h"
+#include <string>
+#include <iostream>
+#include <cmath>
 
-// Command to have the plane reply with its current position and velocity.
-#define COMMAND_RADAR_PING 1
-// Command to have the plane change its velocity.
-#define COMMAND_SET_VELOCITY 2
-// Command to have the plane thread terminate.
-#define COMMAND_EXIT_THREAD 3
+using namespace std;
 
+using std::string;
 // How often the plane should update its position.
 #define POSITION_UPDATE_INTERVAL_SECONDS 1
 // Used internally to identify when the position update timer has fired.
 #define CODE_TIMER 1
 
-typedef struct
-{
+struct Vec3 {
+	Vec3 sum(Vec3 b) {
+		return {x+b.x, y+b.y, z+b.z};
+	}
+	Vec3 absoluteDiff(Vec3 b) {
+		return {abs(x-b.x),abs(y-b.y),abs(z-b.z)};
+	}
+	string print() {
+		return std::to_string(x) + "," + std::to_string(y) + ","
+				+ std::to_string(z);
+	}
+	Vec3 scalarMultiplication(int scalarMultiplier) {
+		return {x*scalarMultiplier, y*scalarMultiplier, z*scalarMultiplier};
+	}
 	int x;
 	int y;
 	int z;
-} Vec3;
+};
```

Comment:
I see that it was necessary to change this from an anonymous struct with typedef to a named struct. For consistency with the rest of the code, I would like to change this to
```
typedef struct Vec3 {
// contents
} Vec3;
```
Since we're in C++ and not raw C, there is very little difference between `struct` and `typedef struct`. However, this will ensure stylistic consistency within the codebase while allowing the computer system to work as expected. See [this stackoverflow answer](https://stackoverflow.com/a/1675446) for an overview of the semantic differences between `struct` and `typedef struct` in C and C++.

Reply:
This is interesting, sounds good to me! Added in b1d9fb7e3deffaf386c7a3ce2963011804976860